cache/lru.go

```diff
@@ -20,27 +34,27 @@ type SizedLRU struct {
 	// SizedLRU will evict items as needed to maintain the total size of the cache
 	// below maxSize.
 	maxSize int64
+	onEvict EvictCallback
```

Comment:
Just one quick note here:
maps in go aren't safe to use from multiple goroutines https://golang.org/doc/faq#atomic_maps
Could we add a mutex (RW?) here and make sure we access the map only after acquiring a lock.

Reply:
Correct, as noted in the comment for `SizedLRU`. Since it's a basic datastructure, I think it should be the job of the client to protect it with locks for more flexibility. `fsCache` does so.